hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/node/TestSCMNodeManager.java

```diff
@@ -855,6 +860,10 @@ public void testScmStatsFromNodeReport()
         String storagePath = testDir.getAbsolutePath() + "/" + dnId;
         StorageReportProto report = TestUtils
             .createStorageReport(dnId, storagePath, capacity, used, free, null);
+        NodeReportProto nodeReportProto = TestUtils.createNodeReport(report);
+        nodeReportHandler.onMessage(
```

Comment:
Shouldn't we find out why the existing NodeReportHandler that is part of the created SCM not getting triggered?

Reply:
@cku328 Thanks for the refactoring. I looked at the description of the test, and the what we are trying to validate. The test is meant to validate the logic to process a NodeReport correctly. It is not meant to validate whether a Node report is processed IF a heartbeat is got. Hence, I am OK with either of your patches - Using nodeReportHandler.onMessage or using nodeManager.processNodeManager. Maybe it is better to use the onMessage to keep it consistent with the other test you recently fixed. what do you think?

Reply:
Thanks @avijayanhwx for the comment, I changed some code so that NodeReport can be sent out with the initial heartbeat. I confirmed the NodeReportHandler in SCM is working properly and it can be triggered to process node reports from multiple nodes.